### DISCUSSION TITLE
Can't deploy anymore. Invalid request: `gitSource` missing required property `repoId`.

### DISCUSSION THREAD

**Discussion by StockBet** [vercel/community#7078] [Help]
### Summary

Everything was working fine, but then this morning I can't deploy anymore.

Getting this error -- Invalid request: `gitSource` missing required property `repoId`.

I can deploy the same repos perfectly fine on Railway.

### Steps to Reproduce

Deploy new project.

**Comment by 65536william** [comment]
I am also getting this error when trying to deploy a project using a repo in one of my organisations. I think it must be related to the fact that I have a personal Vercel account and a Vercel account from my employer, and I have to swap my GitHub authorisation between them every time I want to take advantage of the automatic deployments feature when you push to `main`.

I've opened a support ticket too.

**Reply by Kevin-free** [reply]
@StockBet @65536william I also encountered this problem, how did you solve it?

**Reply by 65536william** [reply]
I haven't solved it, but once you have a Vercel deployment in the dashboard, and it's connected to your repo (yes, it seems to connect even if you see this error), you can add another commit to trigger an automatic deployment and that *should* deploy fine. Or at least it did for me.

**Reply by amyegan** [reply]
If you're switching GitHub authorization between accounts, you may need to disconnect and reconnect the project and repo to update the connection. You can only have one Vercel account connected to a GitHub account, otherwise you would run into login issues. 

If all you need is a basic version of automatic deployment for your personal projects, then you might consider setting up a GitHub Action instead of switching GitHub auth between accounts: https://vercel.com/guides/how-can-i-use-github-actions-with-vercel

**Reply by Kevin-free** [reply]
> If you're switching GitHub authorization between accounts, you may need to disconnect and reconnect the project and repo to update the connection. You can only have one Vercel account connected to a GitHub account, otherwise you would run into login issues.
> 
> If all you need is a basic version of automatic deployment for your personal projects, then you might consider setting up a GitHub Action instead of switching GitHub auth between accounts: https://vercel.com/guides/how-can-i-use-github-actions-with-vercel

@amyegan  Hello, my situation is as follows:

1. I used to be a Pro user, and I had account A (corresponding to Github account A) and Team B (corresponding to Github organization B) in Vercel. And the operation was successful at this time.
2. A few days ago, I downgraded and cancelled Pro (but I should still be within the Pro period), and was prompted to merge all Vercel projects into Team B.

The problem now is:

1. If I operate a project in Github organization B, it fails to deploy, and Vercel prompts: hobby account cannot operate Github organization projects.
2. If I operate a project in Github account A, it fails to deploy, and Vercel prompts: Invalid request: gitSource missing required property repoId.

What should I do now?

**Reply by SarikAnwar01** [reply]
@Kevin-free  i am also facing the same issue did you find any solution for it

**Reply by amyegan** [reply]
What happens if you disconnect the repo from your project and then reconnect it? It was originally authorized under a different team which may be why it is unable to connect at the moment. But if you reset the connection it should start working again.

Please give that a try and let me know whether or not it works for you.

[a comment by an automated account is not shown]
